Source/WebCore/loader/CookieJar.cpp

```diff
@@ -92,11 +92,11 @@ String CookieJar::cookies(Document& document, const URL& url) const
 
     auto pageID = document.pageID();
     std::optional<FrameIdentifier> frameID;
-    if (auto* frame = document.frame())
+    if (RefPtr frame = document.frame())
         frameID = frame->loader().frameID();
 
     std::pair<String, bool> result;
-    if (auto* session = m_storageSessionProvider->storageSession())
+    if (WeakPtr session = Ref { m_storageSessionProvider }->storageSession())
```

Comment:
We should add a `protectedStorageSessionProvider()` since this is done several times.